### PR TITLE
[Feature] Add Starter Pool

### PR DESCRIPTION
**Pull request by DeltaJordan**
I saw someone mention this in the Discord and I decided to give it a shot.  
Also adds a button to "Select None" because during testing I realized how annoying it was to deselect everything manually.

**NOTE: This is NOT tested very extensively due to the ExtraSpace patch issue.** It seems to work just by looking at the starter list in game, but I can't be sure if I messed something else up since I can't get past the Team Name input. Once that issue is resolved I can do more testing.

I hope it's not an issue I keep creating feature pull requests, not too sure what the contributor guidelines are for this project. Just let me know if you would prefer me to make/propose changes through different channels.

### CODE REVIEW

**Comment by End45** [comment]
If you added a "select none" button, maybe a "select all" one would be a good idea too. Accidentally clicking "select none" would be annoying.
And maybe another button to copy the pokémon currently selected for randomization, so you don't have to uncheck two sets of boxes if you want to exclude certain pokémon from appearing altogether.

File: skytemple_randomizer/config.py [hunk 1]
```diff
@@ -497,6 +500,8 @@ def _handle(cls, target, typ: type):
...
... 295, 296, 297, 298, 299, 300, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338, 339, 340, 341, 342, 343, 344, 345, 346, 347, 348, 349, 350, 351, 352, 353, 354, 355, 356, 357, 358, 359, 360, 361, 362, 363, 364, 365, 366, 367, 368, 369, 370, 371, 372, 373, 374, 375, 376, 377, 378, 379, 380, 381, 382, 385, 386, 387, 388, 389, 390, 391, 392, 393, 394, 395, 396, 397, 398, 399, 400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 430, 431, 432, 433, 434, 435, 436, 437, 438, 439, 440, 441, 442, 443, 444, 445, 446, 447, 448, 449, 450, 451, 452, 453, 454, 455, 456, 457, 458, 459, 460, 461, 462, 463, 464, 465, 466, 467, 468, 469, 470, 471, 472, 473, 474, 475, 476, 477, 478, 479, 480, 481, 482, 483, 484, 485, 486, 487, 488, 489, 490, 491, 492, 493, 494, 495, 496, 497, 498, 499, 500, 501, 502, 503, 504, 505, 506, 507, 508, 509, 510, 511, 512, 513, 514, 515, 516, 517, 518, 519, 520, 521, 522, 523, 524, 525, 526, 527, 528, 529, 530, 531, 532, 533, 534, 535, 536]
```
Comment (theCapypara) [inline skytemple_randomizer/config.py:504]: Is this the same as `monsters_enabled`, if so can you put this in a constant and use it for both?

File: skytemple_randomizer/randomizer/util/util.py [hunk 2]
```diff
@@ -121,6 +121,20 @@ def get_allowed_md_ids(conf: RandomizerConfig, with_plus_600=False, *, roster=Ro
         return fun.get_allowed_md_ids(ents, roster)
     return list(ents)
 
+def get_allowed_md_starter_ids(conf: RandomizerConfig, with_plus_600=False, *, roster=Roster.DUNGEON) -> List[u16]:
```
Comment (theCapypara) [inline skytemple_randomizer/randomizer/util/util.py:124]: This should default to Roster.STARTERS instead.

**Review by theCapypara — CHANGES_REQUESTED** [review]
Thank you! See my comments below + I agree with End that there should be a Select All button too.

**Comment by DeltaJordan** [comment]
Maybe the Select All button should be Reset to Default? Otherwise, it will select all the broken/placeholder/etc Pokemon as well.

File: skytemple_randomizer/config.py (hunk 1 above)
Comment (DeltaJordan) [reply skytemple_randomizer/config.py]: See b0d6be0. Adds a constant in `lists.py` named "DEFAULTMONSTERPOOL". I didn't see a better place for constants, so let me know if that works.

File: skytemple_randomizer/randomizer/util/util.py (hunk 2 above)
Comment (DeltaJordan) [reply skytemple_randomizer/randomizer/util/util.py]: Resolved by cce8957

**Comment by DeltaJordan** [comment]
In the latest commit 656dbca I added copy and paste to the Pokemon pools. It might be a bit unclear what they do though, maybe there could be a way to make it clearer?

**Comment by DeltaJordan** [comment]
Reworked the copy functionality and UI:
image

**Comment by End45** [comment]
> Maybe the Select All button should be Reset to Default? Otherwise, it will select all the broken/placeholder/etc Pokemon as well.

Oh yeah, that sounds better

**Comment by DeltaJordan** [comment]
Here's the progress so far on the UI side of things:
image

**Comment by theCapypara** [comment]
If you are done, just hit the reqrequest review button next to my name under "Reviewers" :)

**Review by theCapypara — APPROVED** [review]
Looks good now & seems to work!

**Pull request merged**